### DISCUSSION TITLE
Both overriding and extending colors at the same time doesn't work

### DISCUSSION THREAD

**Discussion by nurbolatk** [tailwindlabs/tailwindcss#12978] [Help] [answered]
Hello, I have a custom design system and I need to configure tailwind in this way:
```js
// tailwind.config.js

const primitives = {
  gold: "#D19C1D",
  royalBlue: "#0F4C81",
  gray: "#CCCCCC",
  purple: "#A020F0",
};

const baseColors = {
  white: "#FFFFFF",
  black: "#000000",
  transparent: "transparent",
  current: "currentColor",
};

/** @type {import('tailwindcss').Config} */
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    textColor: {
      primary: primitives.gold,
      fancy: primitives.purple,
    },
    backgroundColor: {
      primary: primitives.royalBlue,
    },
    extend: {
      colors: {
        ...baseColors,
        gray: primitives.gray,
      },
    },
  },
};

```

So that I can use:
```html
<div className="bg-black text-white">
  <h1 className="text-3xl">Vite + React</h1>
  <div className="card bg-primary">
    <button  className="text-primary">
      count is {count}
    </button>
    <p className="text-gray">
      Edit <code>src/App.tsx</code> and save to test HMR
    </p>
  </div>
  <p>Click on the Vite and React logos to learn more</p>
</div>
```

Notice that our `text-primary` and `bg-primary` classes point to different colors. That is why I am overriding specific style's colors. But at the same time I need some colors that are used in all styles. However, this config doesn't work - the extended colors are missing:

![image](https://github.com/tailwindlabs/tailwindcss/assets/28866825/510350cc-9b88-4114-83d5-073dc473577f)

link to [reproduction](https://codesandbox.io/p/devbox/test-tailwind-rvmzkh?file=%2Fsrc%2Findex.css%3A4%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clssnhbij00073b6i08pqdm8h%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clssnhbij00023b6iwqsquaik%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clssnhbij00043b6iw4s72jsv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clssnhbij00063b6iz6i5iddt%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clssnhbij00023b6iwqsquaik%2522%253A%257B%2522id%2522%253A%2522clssnhbij00023b6iwqsquaik%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clssnhbij00063b6iz6i5iddt%2522%253A%257B%2522id%2522%253A%2522clssnhbij00063b6iz6i5iddt%2522%252C%2522activeTabId%2522%253A%2522clssnmbdu00a73b6h5terteif%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clssnhbij00053b6ic8qlk76r%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clssnmbdu00a73b6h5terteif%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clssnhbij00043b6iw4s72jsv%2522%253A%257B%2522id%2522%253A%2522clssnhbij00043b6iw4s72jsv%2522%252C%2522activeTabId%2522%253A%2522clssnhsga006p3b6i7mr9vuko%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clssnhbij00033b6ifxfzkzkl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clssnhsga006p3b6i7mr9vuko%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clssnhsqq003gdjipacog6wrw%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) 
How can I achieve this?

**Comment by wongjn** [comment, marked as answer]
Consider defining your `textColor` and `backgroundColor` in `theme.extends` rather than `theme`:

```diff
 export default {
   content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
   theme: {
-    textColor: {
-      primary: primitives.gold,
-      fancy: primitives.purple,
-    },
-    backgroundColor: {
-      primary: primitives.royalBlue,
-    },
     extend: {
       colors: {
         ...baseColors,
         gray: primitives.gray,
       },
+      textColor: {
+        primary: primitives.gold,
+        fancy: primitives.purple,
+      },
+      backgroundColor: {
+        primary: primitives.royalBlue,
+      },
     },
   },
 };
```

See this working in [this Tailwind Play](https://play.tailwindcss.com/KMeNdhEnAp?file=config).

**Reply by nurbolatk** [reply]
Thanks! One of my goals was also to remove the default colors that come with tailwind CSS, and this is how I did it:
```js
  theme: {
    colors: {},
    extend: {
      colors: {
        ...baseColors,
        gray: primitives.gray,
      },
      textColor: {
        primary: primitives.gold,
        fancy: primitives.purple,
      },
      backgroundColor: {
        primary: primitives.royalBlue,
      },
    },
  },
```
